src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java

```diff
@@ -34,80 +35,173 @@ public class JSONCodecRegistry {
+    Preconditions.checkNotNull(value, "value must not be null");
+
+    // BUG: needs to handle NULl value
+    var columnMetadata =
+        table.getColumn(column).orElseThrow(() -> new UnknownColumnException(table, column));
+
+    // compiler telling me we need to use the unchecked assignment again like the codecFor does
+    JSONCodec<JavaT, CqlT> codec =
+        JSONCodec.unchecked(internalCodecForToCQL(columnMetadata.getType(), value));
+    if (codec != null) {
+      return codec;
+    }
+    throw new MissingJSONCodecException(table, columnMetadata, value.getClass(), value);
   }
 
+  public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToJSON(
+      TableMetadata table, CqlIdentifier column)
+      throws UnknownColumnException, MissingJSONCodecException {
+
+    Preconditions.checkNotNull(table, "table must not be null");
+    Preconditions.checkNotNull(column, "column must not be null");
+
+    var columnMetadata =
+        table.getColumn(column).orElseThrow(() -> new UnknownColumnException(table, column));
+
+    // compiler telling me we need to use the unchecked assignment again like the codecFor does
+    JSONCodec<JavaT, CqlT> codec =
+        JSONCodec.unchecked(internalCodecForToJSON(columnMetadata.getType()));
+    if (codec != null) {
+      return codec;
+    }
+    throw new MissingJSONCodecException(table, columnMetadata, null, null);
+  }
+
+  /**
+   * Internal only method to find a codec for the specified type and value.
+   *
+   * <p>The return type is {@code JSONCodec<?, ?>} because type erasure means that returning {@code
+   * JSONCodec<JavaT, CqlT>} would be erased. Therefore, we need to use {@link JSONCodec#unchecked}
+   * anyway, which results in this method returning {@code <?, ?>}. However, you are guaranteed that
+   * it will match the types you wanted, due to the call to the codec to test.
+   *
+   * @param targetCQLType
+   * @param javaValue
+   * @return The codec, or `null` if none found.
+   */
+  private static JSONCodec<?, ?> internalCodecForToCQL(DataType targetCQLType, Object javaValue) {
+    // BUG: needs to handle NULl value
+    return CODECS.stream()
+        .filter(codec -> codec.testToCQL(targetCQLType, javaValue))
+        .findFirst()
+        .orElse(null);
+  }
+
+  /**
+   * Same as {@link #internalCodecForToCQL(DataType, Object)}
+   *
+   * @param targetCQLType
+   * @return
+   */
+  private static JSONCodec<?, ?> internalCodecForToJSON(DataType targetCQLType) {
+    return CODECS.stream()
+        .filter(codec -> codec.testToJSON(targetCQLType))
+        .findFirst()
+        .orElse(null);
+  }
+
+  // Boolean
+  public static final JSONCodec<Boolean, Boolean> BOOLEAN =
+      new JSONCodec<>(
+          GenericType.BOOLEAN,
+          DataTypes.BOOLEAN,
+          JSONCodec.ToCQL.unsafeIdentity(),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::booleanNode));
+
   // Numeric Codecs
   public static final JSONCodec<BigDecimal, Long> BIGINT =
```

Comment:
For performance reasons we could also consider only converting FP values into `BigDecimal` JsonNode -- but converting CQL integer values into `long`-valued `JsonNode`.
I think our internal handling can deal with `Integer` and `Long` valued `JsonNode`s and this avoids some of `BigDecimal` overhead (avoids conversion overhead, serialization is faster).

Reply:
added TODO